scripts/stablex/setup_thegraph_data.js

```diff
@@ -0,0 +1,233 @@
+    ],
+  })
+}
+
+async function _submitSolution(name, batchId, solution, solver, batchExchange) {
+  console.log(`Submit "${name}": ${JSON.stringify(solution)}`)
+  const objectiveValue = await batchExchange.submitSolution(
+    batchId,
+    1,
+    solution.owners,
+    solution.touchedorderIds,
+    solution.volumes,
+    solution.prices,
+    solution.tokenIdsForPrice,
+    { from: solver }
+  )
+  console.log(`Transaction for ${name}: ${objectiveValue.tx}`)
+}
+
+async function _deleteOrder({ orderId, user, batchExchange }) {
+  const cancelReceipt = await batchExchange.cancelOrders([orderId], { from: user.address })
+  const events = cancelReceipt.logs.map(log => log.event).join(", ")
+  console.log("Canceled/Deleted order %d. Emitted events: %s", orderId, events)
+}
+
+async function _createOrder({ user, buyToken, sellToken, validUntil, buyAmount, sellAmount, batchExchange }) {
+  // placeOrder(uint16 buyToken, uint16 sellToken, uint32 validUntil, uint128 buyAmount, uint128 sellAmount)
+  console.log(
+    "Create new order for user %s. Buy %s for %s at %s/%s",
+    user.address,
+    buyToken.id,
+    sellToken.id,
+    buyAmount,
+    sellAmount
+  )
+  await batchExchange.placeOrder(buyToken.id, sellToken.id, validUntil, buyAmount, sellAmount, {
+    from: user.address,
+  })
+  const auctionElements = decodeOrdersBN(await batchExchange.getEncodedUserOrders(user.address))
+  const newOrder = auctionElements[auctionElements.length - 1]
+  return {
+    id: auctionElements.length - 1,
+    data: newOrder,
+  }
+}
+
+module.exports = async function(callback) {
+  try {
+    const batchExchange = await BatchExchange.deployed()
+
+    // Prepare user accounts
+    const accounts = await web3.eth.getAccounts()
+    const [user1Address, user2Address] = accounts
+    const user1 = { id: 0, address: user1Address }
+    const user2 = { id: 1, address: user2Address }
+    const solver = user1
+
+    // Get current batch id
+    let batchId = await _getBatchId(batchExchange)
+    console.log("Current batch id: ", batchId)
+
+    // Set user1 as minter of OWL
+    const owl = await _getOwl()
+    await owl.setMinter(user1.address)
+
+    // Mint owl for user1 and user2
+    const amount = web3.utils.toWei("3000")
+    await _mintOWL({ user: user1, owl, amount, batchExchange })
+    await _mintOWL({ user: user2, owl, amount, batchExchange })
+
+    // Create 3 tokens
+    const token1 = await _createAndAddToken(batchExchange)
+    const token2 = await _createAndAddToken(batchExchange)
+    // const token3 = _createAndAddToken(batchExchange)
+
+    // Mint the 2 tokens for the 2 users
+    // const tokens = [owl, token1, token2, token3]
+    const tokens = [token1, token2]
+    tokens.forEach(token => {
+      _mintToken({ token, user: user1, amount, batchExchange })
+      _mintToken({ token, user: user2, amount, batchExchange })
+    })
+
+    // Generate a basic test case
+    const testCase = _getTestCase({ user1, user2, token1, token2 })
```

Comment:
I think  token2 is not used in your example. You are only trading against owl

Reply:
Yes, u right. I was going to create another example, but for now didn't use it.
I'll remove it and add it when/if I evolve the script